xscen/aggregate.py

```diff
@@ -467,3 +491,282 @@ def spatial_mean(
         ds_agg.attrs["cat:processing_level"] = to_level
 
     return ds_agg
+
+
+@parse_config
+def produce_warming_level(
+    ds,
+    wl,
+    indicators,
+    window=20,
+    min_window=None,
+    tas_base_period=["1850", "1900"],
+    ignore_member=False,
+    tas_csv=None,
+    to_level="climatology-warminglevels-{wl}",
+):
+    """
+    Compute the climatology of indicators for a given warming level.
+
+    First, this function calculates the start and end year of a window when the given
+    global warming level is reached from a csv of yearly global mean temperature. Second,
+    the function computes indicators and interrannual mean on this same window. Third,
+    it stack the season and month in different dimensions and adds a dimension horizon for the warming level.
+    The 'horizon' dimension is added in order to facilicate the concatenation with a
+    reference climatological mean before calculating the deltas.
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+      Input dataset with a time dimension and "cat:" attributes.
+    wl: int
+      Warming level.
+      eg. 2 for a global warming level of +2 degree Celsius above the mean temperature of the `tas_ref_period`.
+    indicators:  Union[str, PosixPath, Sequence[Indicator], Sequence[Tuple[str, Indicator]]]
+      Indicators to compute. It will be passed to the `indicators` argument of `xs.compute_indicators`.
+    window: int
+      Size of the window over which the mean tempearture reaches wl and for over which
+      to calculate the climatological mean of the indicators.
+    min_window: int
+      Minimum number of years required for climatological mean to be computed.
+      If left at None, it will be window-2. This is to get a non-nan value for periods
+      that go from december to january (ex. DJF in QS-DEC indicators) because they will have
+      less occurrences than the window size because nans are dropped.
+    tas_base_period: list
+      Base period. The warming is calculated with respect to this period.
+    ignore_member: bool
+      Whether to use the row of the warming_level_csv, even if the member is wrong.
+    tas_csv: str
+      Path to a csv of annual global mean temperature.
+      If None, it will default to data/IPCC_annual_global_tas.csv which was built from
+      the IPCC atlas data ( https://github.com/IPCC-WG1/Atlas/tree/main/datasets-aggregated-regionally/data )
+      and extra data from pilot models of MRCC5 and ClimEx.
+    to_level:
+      The processing level to assign to the output.
+      Use "{wl}" in the string to dynamically include the warming level.
+
+    Returns
+    -------
+    xr.Dataset
+        Warming level dataset
+    """
+    if min_window is None:
+        min_window = window - 2
```

Comment:
Should we change it in `climatological_mean` then? If we don't, all DJFs will be nan...

Reply:
Yeah I left another comment later, but I'd be in favor to find a way to fix that in `climatological_mean`, since that issue is not only related to warming levels.

Could we maybe check `cat:xrfreq` and if it's `AS-xx` that isn't `JAN`, then we use `- 2`? Or do we always use it ?

Reply:
ok for fixing it in `climatological_mean`. 
We would have to check for all frequencies that have a period that contains both dec 31st to Jan 1st. I'm not sure if there is an easy way to check that...

Reply:
added the check in `climatological_mean`